drivers/pinmux/pinmux_stm32.c

```diff
@@ -153,6 +153,79 @@ int stm32_dt_pinctrl_configure(const struct soc_gpio_pinctrl *pinctrl,
 }
 
 #if DT_HAS_COMPAT_STATUS_OKAY(st_stm32f1_pinctrl)
+/* enable remap : modify MAPR and keep the AFIO_MAPR_SWJ_CFG_x */
+#if defined(CONFIG_GPIO_STM32_SWJ_ENABLE)
+/* reset state */
+#define enable_remap(REMAP_PIN) MODIFY_REG(AFIO->MAPR,\
+				(REMAP_PIN | AFIO_MAPR_SWJ_CFG), \
+				(REMAP_PIN | AFIO_MAPR_SWJ_CFG_RESET))
+#elif defined(CONFIG_GPIO_STM32_SWJ_NONJTRST)
+/* released PB4 */
+#define enable_remap(REMAP_PIN) MODIFY_REG(AFIO->MAPR, \
+				(REMAP_PIN | AFIO_MAPR_SWJ_CFG), \
+				(REMAP_PIN | AFIO_MAPR_SWJ_CFG_NOJNTRST))
+#elif defined(CONFIG_GPIO_STM32_SWJ_NOJTAG)
+/* released PB4 PB3 PA15 */
+#define enable_remap(REMAP_PIN) MODIFY_REG(AFIO->MAPR, \
+				(REMAP_PIN | AFIO_MAPR_SWJ_CFG), \
+				(REMAP_PIN | AFIO_MAPR_SWJ_CFG_JTAGDISABLE))
+#elif defined(CONFIG_GPIO_STM32_SWJ_DISABLE)
+/* released PB4 PB3 PA13 PA14 PA15 */
+#define enable_remap(REMAP_PIN) MODIFY_REG(AFIO->MAPR, \
+				(REMAP_PIN | AFIO_MAPR_SWJ_CFG), \
+				(REMAP_PIN | AFIO_MAPR_SWJ_CFG_DISABLE))
+#endif
+
+/* enable remap : modify MAPR and keep the AFIO_MAPR_SWJ_CFG_x */
+#if defined(CONFIG_GPIO_STM32_SWJ_ENABLE)
+/* reset state */
+#define enable_partial_remap(REMAP_PIN, PARTIAL_REMAP) \
+			MODIFY_REG(AFIO->MAPR, \
+				(REMAP_PIN | AFIO_MAPR_SWJ_CFG), \
+				(PARTIAL_REMAP | AFIO_MAPR_SWJ_CFG_RESET))
+#elif defined(CONFIG_GPIO_STM32_SWJ_NONJTRST)
+/* released PB4 */
+#define enable_partial_remap(REMAP_PIN, PARTIAL_REMAP) \
+			MODIFY_REG(AFIO->MAPR, \
+				(REMAP_PIN | AFIO_MAPR_SWJ_CFG), \
+				(PARTIAL_REMAP | AFIO_MAPR_SWJ_CFG_NOJNTRST))
+#elif defined(CONFIG_GPIO_STM32_SWJ_NOJTAG)
+/* released PB4 PB3 PA15 */
+#define enable_partial_remap(REMAP_PIN, PARTIAL_REMAP) \
+			MODIFY_REG(AFIO->MAPR, \
+				(REMAP_PIN | AFIO_MAPR_SWJ_CFG), \
+				(PARTIAL_REMAP | AFIO_MAPR_SWJ_CFG_JTAGDISABLE))
+#elif defined(CONFIG_GPIO_STM32_SWJ_DISABLE)
+/* released PB4 PB3 PA13 PA14 PA15 */
+#define enable_partial_remap(REMAP_PIN, PARTIAL_REMAP) \
+			MODIFY_REG(AFIO->MAPR, \
+				(REMAP_PIN | AFIO_MAPR_SWJ_CFG), \
+				(PARTIAL_REMAP | AFIO_MAPR_SWJ_CFG_DISABLE))
+#endif
+
+/* disable remap : modify MAPR and keep the AFIO_MAPR_SWJ_CFG_x */
+#if defined(CONFIG_GPIO_STM32_SWJ_ENABLE)
+/* reset state */
+#define disable_remap(REMAP_PIN) MODIFY_REG(AFIO->MAPR,\
+				(REMAP_PIN | AFIO_MAPR_SWJ_CFG), \
+				AFIO_MAPR_SWJ_CFG_RESET)
+#elif defined(CONFIG_GPIO_STM32_SWJ_NONJTRST)
+/* released PB4 */
+#define disable_remap(REMAP_PIN) MODIFY_REG(AFIO->MAPR, \
+				(REMAP_PIN | AFIO_MAPR_SWJ_CFG), \
+				AFIO_MAPR_SWJ_CFG_NOJNTRST)
+#elif defined(CONFIG_GPIO_STM32_SWJ_NOJTAG)
+/* released PB4 PB3 PA15 */
+#define disable_remap(REMAP_PIN) MODIFY_REG(AFIO->MAPR, \
+				(REMAP_PIN | AFIO_MAPR_SWJ_CFG), \
+				AFIO_MAPR_SWJ_CFG_JTAGDISABLE)
+#elif defined(CONFIG_GPIO_STM32_SWJ_DISABLE)
+/* released PB4 PB3 PA13 PA14 PA15 */
+#define disable_remap(REMAP_PIN) MODIFY_REG(AFIO->MAPR, \
+				(REMAP_PIN | AFIO_MAPR_SWJ_CFG), \
+				AFIO_MAPR_SWJ_CFG_DISABLE)
+#endif
```

Comment:
What about something more factorized like:
```
if defined(CONFIG_GPIO_STM32_SWJ_ENABLE)
/* reset state */
#define Z_AFIO_REMAP AFIO_MAPR_SWJ_CFG_RESET
#elif defined(CONFIG_GPIO_STM32_SWJ_NONJTRST)
/* released PB4 */
#define Z_AFIO_REMAP AFIO_MAPR_SWJ_CFG_RESET AFIO_MAPR_SWJ_CFG_NOJNTRST
#elif defined(CONFIG_GPIO_STM32_SWJ_NOJTAG)
/* released PB4 PB3 PA15 */
#define Z_AFIO_REMAP AFIO_MAPR_SWJ_CFG_RESET AFIO_MAPR_SWJ_CFG_JTAGDISABLE
#elif defined(CONFIG_GPIO_STM32_SWJ_DISABLE)
/* released PB4 PB3 PA13 PA14 PA15 */
#define Z_AFIO_REMAP AFIO_MAPR_SWJ_CFG_RESET AFIO_MAPR_SWJ_CFG_DISABLE
#endif


#define enable_remap(REMAP_PIN) MODIFY_REG(AFIO->MAPR,\
				           (REMAP_PIN | AFIO_MAPR_SWJ_CFG), \
				           (REMAP_PIN | Z_AFIO_REMAP))

#define enable_partial_remap(REMAP_PIN, PARTIAL_REMAP) \
				MODIFY_REG(AFIO->MAPR, \
					   (REMAP_PIN | AFIO_MAPR_SWJ_CFG), \
					   (PARTIAL_REMAP | Z_AFIO_REMAP))

#define disable_remap(REMAP_PIN) MODIFY_REG(AFIO->MAPR,\
					    (REMAP_PIN | AFIO_MAPR_SWJ_CFG), \
					    Z_AFIO_REMAP)

```

Reply:
ok

Reply:
should Z_AFIO_REMAP be defined based on the **CONFIG_GPIO_STM32_SWJ_xxx** defnition : 
```
#if defined(CONFIG_GPIO_STM32_SWJ_ENABLE)
/* reset state */
#define Z_AFIO_REMAP AFIO_MAPR_SWJ_CFG_RESET
#elif defined(CONFIG_GPIO_STM32_SWJ_NONJTRST)
/* released PB4 */
#define Z_AFIO_REMAP AFIO_MAPR_SWJ_CFG_NOJNTRST
#elif defined(CONFIG_GPIO_STM32_SWJ_NOJTAG)
/* released PB4 PB3 PA15 */
#define Z_AFIO_REMAP AFIO_MAPR_SWJ_CFG_JTAGDISABLE
#elif defined(CONFIG_GPIO_STM32_SWJ_DISABLE)
/* released PB4 PB3 PA13 PA14 PA15 */
#define Z_AFIO_REMAP AFIO_MAPR_SWJ_CFG_DISABLE
#endif
```